integration/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java

```diff
@@ -444,18 +442,21 @@ public void loggerWithTemplate() throws Exception {
   @Description("Processors in global error handlers are intercepted correctly when error is in referenced flow")
   public void globalErrorHandlerWithFlowRef() throws Exception {
     expectedError.expectErrorType("TEST", "EXPECTED");
-
+    CountDownLatch allAftersWereCalled = new CountDownLatch(4);
 
     AtomicInteger afters = new AtomicInteger(0);
 
     AfterWithCallbackInterceptor.callback = (event, thrown) -> {
       afters.incrementAndGet();
+      allAftersWereCalled.countDown();
     };
 
     try {
       flowRunner("flowWithFailingFlowRef").run();
     } finally {
       // The MP in the global error handler is ran twice, once for the called flow and another for the caller flow.
+      // If the afters don't reach 4, this latch will timeout.
```

Comment:
actually, the test will timeout, not the latch.

Reply:
changed